frontend/src/service/ChangeLog.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Update } from '../entity/Update';
+
+type Time = number;
+
+export class ChangeLogService {
+  hasUpdates(): Promise<boolean> {
+    return new Promise(async (resolve, reject) => {
+      const changeLog = await this.getChangeLog();
+      const lastSeenAt = await this.getLastSeenChangeLog();
+      if (!changeLog) {
```

Comment:
Move this before line 9. `getLastSeenChangeLog` is a network request and takes time. If there is no changelog, then there is no point of loading `getLastSeenChangeLog`. This also reduces our server load.

Reply:
@byliuyang, getChangeLog is also a network request. Most probably changeLog will not be empty so i don't think moving [these](https://github.com/short-d/short/pull/560/files#diff-b7dccee6e4430bdac2b4f2d5ce242846R10-R18) lines will not have must intended effect